js/main.js

```diff
@@ -0,0 +1,137 @@
+function renderEntries(entry) { // Creates DOM tree
+  var li = document.createElement('li');
+  var rowDiv = document.createElement('div');
+  rowDiv.className = 'row';
+  li.appendChild(rowDiv);
+  var imgColumnDiv = document.createElement('div');
+  imgColumnDiv.className = 'column-half';
+  rowDiv.appendChild(imgColumnDiv);
+  var img = document.createElement('img');
+  imgColumnDiv.appendChild(img);
+  img.src = entry.photoUrl;
+  var textColumnDiv = document.createElement('div');
+  textColumnDiv.className = 'column-half';
+  rowDiv.appendChild(textColumnDiv);
+  var title = document.createElement('span');
+  var text = document.createElement('p');
+  textColumnDiv.appendChild(title);
+  textColumnDiv.appendChild(text);
+  var entryTitle = document.createTextNode(entry.title);
+  title.appendChild(entryTitle);
+  var entryNotes = document.createTextNode(entry.notes);
+  text.appendChild(entryNotes);
+  return li;
+}
+
+var ul = document.querySelector('ul');
+
+window.addEventListener('DOMContentLoaded', appendDomContent); // Appends DOM trees for new entries
+
+function appendDomContent(event) {
+  for (var i = 0; i < data.entries.length; i++) {
+    if (data.entries[i] !== null) {
+      ul.appendChild(renderEntries((data.entries[i])));
+    }
+  }
+}
+
+var noEntry = document.querySelector('.no-entry'); // Displays no entry text when no entries have been entered
+
+if (data.entries.length === 0) {
+  noEntry.className = 'no-entry';
+}
+
+var display = document.querySelectorAll('.display');
+
+window.addEventListener('click', click);
+
+var inputCheck = document.querySelectorAll('input');
+var textarea = document.querySelector('textarea');
+
+function click(event) { // switches view between pages
+  var dataView = event.target.getAttribute('data-view');
+  if (event.target.matches('.view-entries')) {
+    for (var i = 0; i < display.length; i++) {
+      display[i].className = 'display hidden';
+      if (display[i].getAttribute('data-view') === dataView) {
+        display[i].className = 'display';
+        data.view = 'entries';
+      }
+    }
+  } else if (event.target.matches('.new-button')) {
+    for (i = 0; i < display.length; i++) {
+      display[i].className = 'display hidden';
+      if (display[i].getAttribute('data-view') === dataView) {
+        display[i].className = 'display';
+        data.view = 'entry-form';
+      }
+    }
+  } else if (event.target.matches('.submit')) { // checks if any inputs on the entry form are empty
+    for (i = 0; i < 1; i++) {
+      for (var j = 0; j < inputCheck.length; j++) {
+        if (inputCheck[j].value.length === 0 || textarea.value.length === 0) {
+          return;
+        }
+      }
+    }
+    for (i = 0; i < display.length; i++) { // switches view when an entry is saved
+      display[i].className = 'display hidden';
+      if (display[i].getAttribute('data-view') === dataView) {
+        display[i].className = 'display';
+        data.view = 'entries';
+      }
+    }
+  }
+}
```

Comment:
I would create a separate function for each of the `buttons` which are clicked on compartmentalize the code more.